packages/store/test/test-pattern-limits.js

```diff
@@ -0,0 +1,243 @@
+// @ts-check
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { makeCopyBag, makeCopyMap, makeCopySet } from '../src/keys/checkKey.js';
+import {
+  fit,
+  matches,
+  M,
+  defaultLimits,
+} from '../src/patterns/patternMatchers.js';
+import '../src/types.js';
+
+/**
+ * @typedef MatchTest
+ * @property {Passable} specimen
+ * @property {Pattern[]} yesPatterns
+ * @property {[Pattern, RegExp|string][]} noPatterns
+ */
+
+/** @type {MatchTest[]} */
+const limitTests = harden([
+  // decimalDigitsLimit
+  {
+    specimen: 379n,
+    yesPatterns: [
+      M.bigint(),
+      M.bigint(harden({ decimalDigitsLimit: 3 })),
+      M.nat(),
+      M.nat(harden({ decimalDigitsLimit: 3 })),
+    ],
+    noPatterns: [
+      [
+        M.bigint({ decimalDigitsLimit: 2 }),
+        'bigint "[379n]" must not have more than 2 digits',
+      ],
+      [
+        M.nat({ decimalDigitsLimit: 2 }),
+        'bigint "[379n]" must not have more than 2 digits',
+      ],
+    ],
+  },
+  {
+    specimen: -379n,
+    yesPatterns: [M.bigint(), M.bigint(harden({ decimalDigitsLimit: 3 }))],
```

Comment:
The sneaky reason I put "Limit" in all the property names is in case we ever want to grow this optional bag of limits into a more general options bag containing other options. But in any case, by the hardening rules I would still write your expression as
```js
M.bigint(limits(harden({ decimalDigits: 3 })))
```
because I don't imagine teaching the harden static checker about `limits` as a special case.

In any case, I took these as suggestions to keep in mind as we evolve this API rather than changes needed for this PR, so resolving. If I misunderstood, please unresolve.